runhouse/resources/hardware/cluster.py

```diff
@@ -664,39 +664,37 @@ def check_server(self, restart_server=True):
             )
 
         if not self.client:
-            try:
+            self.connect_server_client()
+
+        try:
+            self.client.check_server()
```

Comment:
main update is still checking the server connection even if client is set, in case anything goes down in between check servers.

also reconnecting server client in the restart_server loop, in case of using an old client in the previous connect server client (in the case of loading down stale ips and the prev client is formed with that)